### PR TITLE
doc:add reference to CSI snapshot deployment guide

### LINKED ISSUE #4100: Missing CRDs of the CSI Snapshotter in the "deploy" Ressources
# Describe the bug #

The `csi-snapshotter` container got the following errors, after deploying the [deploy](https://github.com/ceph/ceph-csi/tree/v3.8.1/deploy) ressources of `fs` and `rbd`:

```
csi-snapshotter E0824 10:01:40.186432       1 reflector.go:140] github.com/kubernetes-csi/external-snapshotter/client/v6/informers/externalversions/factory.go:117: Failed to watch *v1.VolumeSnapshotContent: failed to list *v1.VolumeSnapshotContent: the server could not find the requested resource (get volumesnapshotcontents.snapshot.storage.k8s.io)
csi-snapshotter W0824 10:02:03.644213       1 reflector.go:424] github.com/kubernetes-csi/external-snapshotter/client/v6/informers/externalversions/factory.go:117: failed to list *v1.VolumeSnapshotClass: the server could not find the requested resource (get volumesnapshotclasses.snapshot.storage.k8s.io)
csi-snapshotter E0824 10:02:03.644354       1 reflector.go:140] github.com/kubernetes-csi/external-snapshotter/client/v6/informers/externalversions/factory.go:117: Failed to watch *v1.VolumeSnapshotClass: failed to list *v1.VolumeSnapshotClass: the server could not find the requested resource (get volumesnapshotclasses.snapshot.storage.k8s.io)
csi-snapshotter W0824 10:02:37.337592       1 reflector.go:424] github.com/kubernetes-csi/external-snapshotter/client/v6/informers/externalversions/factory.go:117: failed to list *v1.VolumeSnapshotContent: the server could not find the requested resource (get volumesnapshotcontents.snapshot.storage.k8s.io)
```

Root cause: Missing CRDs of the CSI Snapshotter. They can be found here: https://github.com/kubernetes-csi/external-snapshotter/tree/v6.1.0/client/config/crd.

You have to use the CRDs that fit to the deployed `csi-snapshotter` version that is deployed.
E.g. vor ceph-csi version `v3.8.1`, the csi-snapshotter version `v6.1.0` is used.

Everything will be resolved, if you apply the CRDs.

# Environment details #

- Image/version of Ceph CSI driver : v3.8.1
- Kubernetes cluster version : v1.25.12
- Ceph cluster version : v16.2.7 (Pacific)

# Steps to reproduce #

1. Apply the resources in  the [deploy/cephfs/kubernetes](https://github.com/ceph/ceph-csi/tree/v3.8.1/deploy/cephfs/kubernetes) directory (rbd also possible)
2. Check the logs of the `csi-snapshotter`
3. See errors

# Expected behavior #

The Kuberntes resources in the [deploy](https://github.com/ceph/ceph-csi/tree/v3.8.1/deploy) directory must contain the needed CRDs for the `csi-snapshotter`.

### PR DESCRIPTION
**Pull request by iPraveenParihar**
# Describe what this PR does #

This change adds reference to CSI snapshot CRDs 
deployment guide in deploy-cephfs.md and deploy-rbd.md

Closes: #4100

### CODE REVIEW

File: docs/deploy-cephfs.md [hunk 1]
```diff
@@ -151,6 +151,10 @@ for more information.
 kubectl create -f ../../ceph-conf.yaml
 ```
 
+**Deploy CSI Snapshot:**
+
+If you intend to use the snapshot functionality in your Kuberenetes cluster, please refer [deployment guide](./snap-clone.md#prerequisite)
```
Comment (Rakshith-R) [inline docs/deploy-cephfs.md]:
```suggestion
**Deploy prerequisites for CSI Snapshot:**

If you intend to use the snapshot functionality in your Kuberenetes cluster, please refer to [snap-clone.md](./snap-clone.md#prerequisite) 
```

**Review by Rakshith-R — CHANGES_REQUESTED** [review]

File: docs/deploy-cephfs.md [hunk 2]
```diff
@@ -151,6 +151,11 @@ for more information.
 kubectl create -f ../../ceph-conf.yaml
 ```
 
+**Deploy prerequisites for  CSI Snapshot:**
```
Comment (Madhu-1) [inline docs/deploy-cephfs.md]:
```suggestion
**Deploy prerequisites for CSI Snapshot:**
```

File: docs/deploy-cephfs.md [hunk 3]
```diff
@@ -151,6 +151,11 @@ for more information.
...
+
+If you intend to use the snapshot functionality in your Kuberenetes cluster,
```
Comment (Madhu-1) [inline docs/deploy-cephfs.md]:
```suggestion
If you intend to use the snapshot functionality in  Kubernetes cluster,
```

**Review by Madhu-1 — CHANGES_REQUESTED** [review]

**Review by Madhu-1 — DISMISSED** [review]

File: docs/deploy-cephfs.md [hunk 4]
```diff
@@ -151,6 +151,11 @@ for more information.
 kubectl create -f ../../ceph-conf.yaml
 ```
 
+**Deploy prerequisites for CSI Snapshot:**
+
+If you intend to use the snapshot functionality in Kuberenetes cluster,
```
Comment (Madhu-1) [inline docs/deploy-cephfs.md]: still you have typo in Kubernetes

Comment (iPraveenParihar) [reply docs/deploy-cephfs.md]: Ahh, missed it!

**Review by Rakshith-R — DISMISSED** [review]

**Review by riya-singhal31 — DISMISSED** [review]

**Review by Madhu-1 — DISMISSED** [review]

**Comment by Rakshith-R** [comment]
@mergifyio queue

**Comment by Rakshith-R** [comment]
@mergifyio refresh

**Comment by Rakshith-R** [comment]
@mergifyio rebase

**Comment by Madhu-1** [comment]
@Mergifyio requeue default

**Comment by Madhu-1** [comment]
@Mergifyio refresh

**Review by Madhu-1 — DISMISSED** [review]

**Review by Rakshith-R — DISMISSED** [review]

**Comment by Madhu-1** [comment]
@mergifyio queue

**Comment by Madhu-1** [comment]
@mergifyio refresh

**Comment by Madhu-1** [comment]
@mergifyio rebase

**Comment by Madhu-1** [comment]
@Mergifyio rebase

**Comment by Madhu-1** [comment]
@Mergifyio dequeue

**Comment by Rakshith-R** [comment]
@Mergifyio unqueue

**Comment by Rakshith-R** [comment]
@mergifyio unqueue

**Comment by Rakshith-R** [comment]
@iPraveenParihar please rebase this pr manually after this pr https://github.com/ceph/ceph-csi/pull/4124 merges.

**Review by Rakshith-R — APPROVED** [review]

**Comment by Rakshith-R** [comment]
@Mergifyio requeue

**Comment by Rakshith-R** [comment]
@Mergifyio queue

**Review by Madhu-1 — APPROVED** [review]

**Comment by Madhu-1** [comment]
@Mergifyio queue

**Pull request merged**